knowhere/index/vector_index/IndexDiskANN.cpp

```diff
@@ -373,8 +373,8 @@ IndexDiskANN<T>::Query(const DatasetPtr& dataset_ptr, const Config& config, cons
 
     GET_TENSOR_DATA_DIM(dataset_ptr);
     auto query = static_cast<const T*>(p_data);
-    auto p_id = static_cast<int64_t*>(malloc(sizeof(int64_t) * k * rows));
```

Comment:
Check this response, maybe it's not a good idea?
https://github.com/milvus-io/knowhere/pull/298#discussion_r926881741

Reply:
In #350 konwhere has replaced malloc/free with new/delete. So it may not be a problem.